go-controller/pkg/node/gateway_iptables.go

```diff
@@ -79,6 +79,18 @@ func deleteIptRules(rules []nodeipt.Rule) error {
 	return nodeipt.DelRules(rules)
 }
 
+// ensureChain ensures that a chain exists within a table
+func ensureChain(table, chain string) error {
```

Comment:
I think we already have a util for this somewhere no?
I thought insertIpt does chain creation automatically... how is this crash transient and goes away eventually?

Reply:
I couldnt find any.

If its a rule for a chain then itll auto create the chain, but if its a jump, it wont create the chain you jump to.
I assume it hit a rule during init startup that then created the chain but the error inserting the jump rule happened earlier and following reboot of the container, the new chain is there.

Reply:
could you use the IPTablesHelper function `NewChain()` similar to https://github.com/ovn-org/ovn-kubernetes/blob/244efcf34e99780cab97793392d76e6e2ed77807/go-controller/pkg/node/iptables/iptables.go#L108-L111

Reply:
> 
> If its a rule for a chain then itll auto create the chain, but if its a jump, it wont create the chain you jump to. I assume it hit a rule during init startup that then created the chain but the error inserting the jump rule happened earlier and following reboot of the container, the new chain is there.

yea that's it. thanks martin for rca

Reply:
@JacobTanenbaum yea

```
func addChaintoTable(ipt util.IPTablesHelper, tableName, chain string) {
	if err := ipt.NewChain(tableName, chain); err != nil {
		klog.V(5).Infof("Chain: \"%s\" in table: \"%s\" already exists, skipping creation: %v", chain, tableName, err)
	}
}
```
he is indirectly using that